federation/cluster/common.sh

```diff
@@ -24,7 +24,14 @@
 
 : "${KUBE_ROOT?Must set KUBE_ROOT env var}"
 
+# Provides the $KUBERNETES_PROVIDER variable and detect-project function
+source "${KUBE_ROOT}/cluster/kube-util.sh"
+
+# Populates $PROJECT
+detect-project
+
 FEDERATION_IMAGE_REPO_BASE=${FEDERATION_IMAGE_REPO_BASE:-'gcr.io/google_containers'}
+FEDERATION_PUSH_REPO_BASE=${FEDERATION_PUSH_REPO_BASE:-gcr.io/${PROJECT}}
```

Comment:
Fair enough.  Best to fail explicitly in that case (e.g. detect whether there is a ':' in what comes back from detect-project, and print an error message telling the user what to do).